ui/client/containers/NussknackerInitializer.js

```diff
@@ -54,26 +54,31 @@ class NussknackerInitializer extends React.Component {
   componentDidMount() {
     //It looks like callback hell.. Can we do it better?
     HttpService.fetchSettings().then(settingsResponse => {
-      this.props.actions.assignSettings(settingsResponse.data)
-      this.authenticationStrategy(settingsResponse.data.authentication).then(response => {
+      const settings = settingsResponse.data
+      this.props.actions.assignSettings(settings)
+
+      this.authenticationStrategy(settings.authentication).then(response => {
         HttpService.fetchLoggedUser().then(userResponse => {
           this.props.actions.assignUser(userResponse.data)
           this.setState({initialized: true})
-        }).catch(this.httpErrorHandler)
+        }).catch(this.httpErrorHandler, _.isEqual(settings.backend, OAUTH2_BACKEND))
```

Comment:
shouldn't it behave differently for non oauth backend? after your change I assume that we won't handle it at all...

Reply:
For BasicAuth it will behave in different way - will show page with information about non auth user (for BasicAuth there is no redirect).